pykafka/simpleconsumer.py

```diff
@@ -206,6 +206,7 @@ def _update(self):
         with self._update_lock:
             self._cluster.update()
             self._setup_partitions_by_leader()
+            self._discover_offset_manager()
```

Comment:
Ugh, that's nasty. I thought that originally, `_cluster.update()` was supposed to re-connect the existing `Broker` instance, not replace it, so that it was ok to reference that from `Partition`. Did we break that at some point?